### ISSUE TITLE
SFTP support?

### ISSUE THREAD

**Issue by REEMERGEMEDIA** [morris/vinyl-ftp#7]
Are there any plans to implement SFTP?

Thanks.

**Comment by morris** [comment]
No, not currently. I think https://github.com/mscdex/ssh2-streams would provide most of the functionality, so maybe an abstraction of FTP/SFTP inside of vinyl-ftp is viable. However, I think this should go in a separate module `vinyl-sftp` because FTP and SFTP don't seem to be related too much. (I might be wrong about that, though.)

**Comment by droplab** [comment]
+1

**Comment by charlesbodman** [comment]
+1

**Comment by thomaslindstrom** [comment]
+1

**Comment by krispy1298** [comment]
+1

**Comment by pyrsmk** [comment]
+1

**Comment by fgilio** [comment]
+1 this would  be an awesome feature. I use SFTP whenever possible, and some servers even require to use only STFP

**Comment by webcaetano** [comment]
+1

**Comment by skrzyszewski** [comment]
+1

**Comment by levymetal** [comment]
+1

**Comment by OutThisLife** [comment]
+1

**Comment by travisholliday** [comment]
+1

**Comment by MichaelTSS** [comment]
+1

**Comment by christophwolff** [comment]
+1

**Comment by Leagnus** [comment]
+1

**Comment by bennyp123** [comment]
me too, please

**Comment by MarkusDiehl** [comment]
+1

**Comment by pyrsmk** [comment]
Currently, I'm using [gulp-sftp](https://github.com/gtg092x/gulp-sftp) in addition to vinyl-ftp ;)

**Comment by nsteiner** [comment]
+1

**Comment by wujekbogdan** [comment]
+1

**Comment by bmango** [comment]
+1

**Comment by kenanfallon** [comment]
+1

**Comment by Ju1jan** [comment]
+1

**Comment by Marak** [comment]
Have working implementation with `ssh2`:

https://github.com/bigcompany/hook.io-vfs/blob/master/lib/adapters/sftp.js

https://github.com/bigcompany/hook.io-vfs/blob/master/examples/sftp-readFile.js
https://github.com/bigcompany/hook.io-vfs/blob/master/examples/sftp-writeFile.js

Is coupled to multi-cloud virtual file-system. Feel free to use.

**Comment by morris** [comment]
cool!!

**Comment by JacobDB** [comment]
+1

**Comment by Marcel-G** [comment]
+1

**Comment by thibaudalt** [comment]
+1

**Comment by tiefenb** [comment]
+1

**Comment by TCB13** [comment]
+1!

**Comment by alex-steinberg** [comment]
+1

**Comment by hescano** [comment]
+1

**Comment by JacobDB** [comment]
I was just looking at the documentation, and saw there was a parameter for "secured." Is that for SFTP? Going to try and test it later today.

![image](https://cloud.githubusercontent.com/assets/1123997/23076252/3c1b3bda-f505-11e6-97b4-caa45b9c6896.png)

**Comment by naymaeu** [comment]
+1

**Comment by zethodderskov** [comment]
+1

**Comment by Lichtjaeger** [comment]
+1

**Comment by lmammino** [comment]
+1

**Comment by Voltokk** [comment]
+1

**Comment by thomjjames** [comment]
+1

**Comment by balya** [comment]
+1

**Comment by fransstudio2** [comment]
+1

**Comment by onezerodigits** [comment]
![confused](https://user-images.githubusercontent.com/111835/42325813-5434e6a0-8035-11e8-94b4-caacc04ce167.gif)

**Comment by Marak** [comment]
I've had this working in production for some time. The base of the code is using `ssh2` and https://github.com/bigcompany/hook.io-vfs/blob/master/lib/adapters/sftp.js.

If anyone wants this enough to pay me to build a new separate library `vinyl-sftp` let me know and I should be able to get it built.

**Comment by rvducay** [comment]
+1
